frontend/packages/data-portal/app/routes/depositions.$id.tsx

```diff
@@ -192,42 +201,148 @@ export default function DepositionByIdPage() {
+              datasetCounts={
+                groupBy === GroupByOption.DepositedLocation
+                  ? groupedData.datasets.reduce(
+                      (acc, dataset) => {
+                        acc[dataset.id] = {
+                          runCount: dataset.runCount,
+                          annotationCount: dataset.annotationCount,
+                          tomogramRunCount: dataset.tomogramRunCount,
+                        }
+                        return acc
+                      },
+                      {} as Record<
+                        number,
+                        {
+                          runCount: number
+                          annotationCount: number
+                          tomogramRunCount: number
+                        }
+                      >,
+                    )
+                  : undefined
+              }
+            />
           ) : (
             <DatasetsTable />
           ),
 
-          totalCount: match({ isExpandDepositions, tab })
+          totalCount: match({ isExpandDepositions, tab, groupBy })
             .with(
-              { isExpandDepositions: true, tab: DepositionTab.Annotations },
+              { isExpandDepositions: true, groupBy: GroupByOption.Organism },
+              () => groupedData.organisms.length,
+            )
+            .with(
+              {
+                isExpandDepositions: true,
+                groupBy: GroupByOption.DepositedLocation,
+              },
+              () => groupedData.datasets.length,
+            )
+            .with(
+              { isExpandDepositions: true, tab: DataContentsType.Annotations },
               () => annotationsCount,
             )
             .with(
-              { isExpandDepositions: true, tab: DepositionTab.Tomograms },
+              { isExpandDepositions: true, tab: DataContentsType.Tomograms },
               () => tomogramsCount,
             )
             .otherwise(() => totalDatasetsCount),
 
           // TODO replace annotations and tomograms with filtered counts
-          filteredCount: match({ isExpandDepositions, tab })
+          filteredCount: match({ isExpandDepositions, tab, groupBy })
+            .with(
+              { isExpandDepositions: true, groupBy: GroupByOption.Organism },
+              () => groupedData.organisms.length,
+            )
+            .with(
+              {
+                isExpandDepositions: true,
+                groupBy: GroupByOption.DepositedLocation,
+              },
+              () => groupedData.datasets.length,
+            )
             .with(
-              { isExpandDepositions: true, tab: DepositionTab.Annotations },
+              { isExpandDepositions: true, tab: DataContentsType.Annotations },
               () => annotationsCount,
             )
             .with(
-              { isExpandDepositions: true, tab: DepositionTab.Tomograms },
+              { isExpandDepositions: true, tab: DataContentsType.Tomograms },
```

Comment:
This component is quite a bit over 200 lines, but I'm not sure if there's much to be done about it.  It looks like there are already subcomponents broken out.  Would some util functions help?